profiling/src/pcntl.rs

```diff
@@ -0,0 +1,118 @@
+use crate::bindings::{
+    datadog_php_install_handler, datadog_php_zif_handler, zend_execute_data, zend_long, zval,
+    InternalFunctionHandler,
+};
+use crate::{RequestLocals, PROFILER, REQUEST_LOCALS};
+use log::{error, warn};
+use std::cell::RefMut;
+use std::ffi::CStr;
+use std::mem::{forget, swap};
+
+static mut PCNTL_FORK_HANDLER: InternalFunctionHandler = None;
+static mut PCNTL_RFORK_HANDLER: InternalFunctionHandler = None;
+static mut PCNTL_FORKX_HANDLER: InternalFunctionHandler = None;
+
+fn handle_child_fork(mut locals: RefMut<RequestLocals>, retval: &mut zval) {
+    let result: Result<zend_long, _> = retval.try_into();
+    match result {
+        Ok(pid) => {
+            // The child gets pid of 0. For now, stop profiling for safety.
+            if pid == 0 {
+                match PROFILER.lock() {
```

Comment:
May that not ... just block indefinitely when the lock is held by the other thread at time of forking? At least docs aren't mentioning mutex poisoning in context of forking.

Reply:
Yes, that can happen in theory. In practice, I don't support ZTS and the profiler is only used on the PHP thread (caveat, it's also called by the tracer when a trace finishes, so please don't do that in from a non-PHP thread 😅 ).

But, that doesn't mean I'm in the clear. I'm not sure about all of the internals of the objects that the Profiler stores, for instance.

Reply:
Why does it need a mutex at all, when it's only used on a single thread? Or is this in anticipation of future ZTS support?

Reply:
1. Yes, it's in anticipation of ZTS support. Even if it's not a critical part of the business, I think it's smart to have a plan there.
2. Having a global mutable variable like this requires safe abstractions in Rust or using unsafe code. Given the above point, it made sense to me (at the time anyway) to just go ahead and use the mutex.